src/test/java/bio/terra/service/snapshot/flight/SnapshotAuthzIamStepTest.java

```diff
@@ -104,4 +111,36 @@ void testDoAndUndoStepWithDUOS() throws InterruptedException {
     assertThat(undoResult.getStepStatus(), equalTo(StepStatus.STEP_RESULT_SUCCESS));
     verify(iamService).deleteSnapshotResource(TEST_USER, SNAPSHOT_ID);
   }
+
+  @Test
+  void testDoAndUndoWithSnapshotFirecloudGroup() throws InterruptedException {
+    workingMap.put(
+        SnapshotWorkingMapKeys.SNAPSHOT_FIRECLOUD_GROUP_EMAIL, SNAPSHOT_FIRECLOUD_GROUP_EMAIL);
+    when(flightContext.getWorkingMap()).thenReturn(workingMap);
+    overrideSnapshotRequestMode(SnapshotRequestContentsModel.ModeEnum.BYREQUESTID);
+    var expectedPolicies =
+        new SnapshotRequestModelPolicies().addReadersItem(SNAPSHOT_FIRECLOUD_GROUP_EMAIL);
+    Map<IamRole, String> expectedPoliciesMap = new HashMap<>();
+    expectedPoliciesMap.put(IamRole.READER, SNAPSHOT_FIRECLOUD_GROUP_EMAIL);
+    when(iamService.createSnapshotResource(TEST_USER, SNAPSHOT_ID, expectedPolicies))
+        .thenReturn(expectedPoliciesMap);
+    step =
+        new SnapshotAuthzIamStep(
+            iamService, snapshotService, snapshotRequestModel, TEST_USER, SNAPSHOT_ID);
+    StepResult doResult = step.doStep(flightContext);
+    assertThat(doResult.getStepStatus(), equalTo(StepStatus.STEP_RESULT_SUCCESS));
+    Map<IamRole, String> workingMapPolicies =
+        flightContext
+            .getWorkingMap()
```

Comment:
This is calling the mock to get a value that we already have in scope
```suggestion
        workingMap
```